lua/harpeek/init.lua

```diff
@@ -75,29 +79,40 @@ local function format_item(path, index)
     end
 
     -- TODO: This logic is pretty opaque. I could probably do something nicer.
-    local format = ''
-    if Harpeek._settings.format == 'filename' then
-        format = vim.fn.fnamemodify(path, ':t') .. suffix
+    local postfix = ''
+    if format == 'filename' then
+        postfix = vim.fn.fnamemodify(path, ':t') .. suffix
     else
         local relative = vim.fn.fnamemodify(path .. suffix, ':.')
         if #relative == 0 then
-            format = '.'
-        elseif Harpeek._settings.format == 'relative' then
-            format = relative
-        elseif Harpeek._settings.format == 'shortened' then
-            format = vim.fn.pathshorten(vim.fn.fnamemodify(path, ':.')) .. suffix
+            postfix = '.'
+        elseif format == 'relative' then
+            postfix = relative
+        elseif format == 'shortened' then
+            postfix = vim.fn.pathshorten(vim.fn.fnamemodify(path, ':.')) .. suffix
         end
     end
 
-    return index .. ' ' .. format
+    return index .. ' ' .. postfix
 end
 
-function Harpeek.open()
+---@param opts harpeek.settings?
+function Harpeek.open(opts)
+    if not opts then
+        Harpeek._open_opts = Harpeek._settings
+    end
+    Harpeek._open_opts = vim.tbl_extend('force', Harpeek._settings, opts)
+
     local contents = {}
     local longest_line = 0
     local list = ext.get_list()
+    if #list == 0 then
+        vim.notify("No marks")
```

Comment:
Personally I don't like the idea of bailing if there's no items. Some people may just open the window by default in their config then every time they open in a new dir it would bail out and sent an annoying notification.

Reply:
What do you think if i add this behavior
![image](https://github.com/WolfeCub/harpeek.nvim/assets/90571229/9a6ffdcf-e878-4917-b9cb-e744f9ef5592)

Reply:
The only thing that i don't like is number before the message.
If it's ok for you i will think how to do it without an index

Reply:
```lua
if #list == 0 then
    contents = {"No marks"}
    longest_line = contents[1]:len()
end
```

Reply:
![image](https://github.com/WolfeCub/harpeek.nvim/assets/90571229/5495586f-8427-4c06-81cd-0aa5301ea86f)

Reply:
Yeah I don't mind this approach. Thanks for being flexible :)

Reply:
done